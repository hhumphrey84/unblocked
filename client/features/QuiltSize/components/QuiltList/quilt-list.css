@import '../../../../assets/mixins.css';

.list {
    list-style: none;
    padding: 0;
    margin: 0;

    &:focus {
        /* TODO: alternate to outline */
        outline: none;
    }
}

.list-title {
    @mixin usability-hidden;
}
