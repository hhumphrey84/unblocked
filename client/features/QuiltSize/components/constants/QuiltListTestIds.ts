enum QuiltOptionTestIds {
    WRAPPER = 'QUILT_LIST',
    LIST = 'QUILT_LIST__LIST',
    TITLE = 'QUILT_LIST__TITLE',
    OPTION = 'QUILT_LIST__OPTION',
}

export default QuiltOptionTestIds;
