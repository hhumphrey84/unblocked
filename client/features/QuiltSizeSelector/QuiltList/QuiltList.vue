<template>
    <div>
        <p id="quiltListLabel">Select a quilt list:</p>
        <ul
            role="listbox"
            tabindex="0"
            aria-labelledby="quiltListLabel"
        >
            <quilt-size-option
                v-for="option of options"
                :key="option.id"
                :data="option"
                :is-selected="option.id === selectedOptionId"
                :on-click="onOptionClick"
            />
        </ul>
    </div>
</template>

<script>
    import QuiltSizeOption from '../QuiltSizeOption/QuiltSizeOption';

    const components = {
        QuiltSizeOption,
    };

    const props = {
        options: {
            type: Array,
            required: true,
        },
        selectedOptionId: {
            type: String,
            required: true,
        },
        onOptionClick: {
            type: Function,
            required: true,
        },
    };

    export default {
        name: 'QuiltList',
        components,
        props,
    };
</script>
