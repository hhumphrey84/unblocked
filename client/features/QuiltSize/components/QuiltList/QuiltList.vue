<template>
    <div
        :data-test-id="testIds.WRAPPER"
        class="wrapper"
    >
        <p
            id="quiltListLabel"
            :data-test-id="testIds.TITLE"
            class="list-title"
        >
            Select a quilt list:
        </p>
        <ul
            :data-test-id="testIds.LIST"
            class="list"
            role="listbox"
            tabindex="0"
            aria-labelledby="quiltListLabel"
        >
            <quilt-size-option
                v-for="option of options"
                :key="option.id"
                :data="option"
                :data-test-id="testIds.OPTION"
                :is-selected="option.id === selectedOptionId"
                :on-click="onOptionClick"
            />
        </ul>
    </div>
</template>

<style scoped src="./quilt-list.css"></style>

<script>
    import QuiltSizeOption from '../QuiltSizeOption/QuiltSizeOption';
    import { QUILT_LIST_TEST_IDS } from '../../quilt-size.constants';

    const components = {
        QuiltSizeOption,
    };

    const props = {
        options: {
            type: Array,
            required: true,
        },
        selectedOptionId: {
            type: String,
            required: true,
        },
        onOptionClick: {
            type: Function,
            required: true,
        },
    };

    const data = {
        testIds: QUILT_LIST_TEST_IDS,
    };

    export default {
        name: 'QuiltList',
        components,
        props,
        data: () => data,
    };
</script>
