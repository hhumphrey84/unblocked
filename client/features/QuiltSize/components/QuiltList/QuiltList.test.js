import { shallowMount } from '@vue/test-utils';
import QuiltList from './QuiltList.vue';
import QuiltListTestIds from '../constants/QuiltListTestIds';

const options = [
    { id: '0' },
    { id: '1' },
    { id: '2' },
];

const selectedOptionId = options[1].id;

const wrapper = shallowMount(QuiltList, {
    propsData: {
        options,
        selectedOptionId,
        onOptionClick: () => null,
    }
})

describe('QuiltList', () => {
    test('renders properly', () => {
        expect(wrapper.html()).toMatchSnapshot();
    });

    test('should contain a title', () => {
        const title = wrapper.find(`[data-test-id=${QuiltListTestIds.TITLE}]`);

        expect(title.exists()).toBe(true);
    });

    test('should contain a list', () => {
        const list = wrapper.find(`[data-test-id=${QuiltListTestIds.LIST}]`);

        expect(list.exists()).toBe(true);
    });

    test('the id of the title should match the aria-labelledby value', () => {
        const title = wrapper.find(`[data-test-id=${QuiltListTestIds.TITLE}]`);
        const list = wrapper.find(`[data-test-id=${QuiltListTestIds.LIST}]`);

        expect(title.attributes('id')).toBe(list.attributes('aria-labelledby'));
    });

    test('should render options equal to the length of props.options', () => {
        const options = wrapper.findAll(`[data-test-id=${QuiltListTestIds.OPTION}]`);

        expect(options.length).toBe(options.length);
    });
});
