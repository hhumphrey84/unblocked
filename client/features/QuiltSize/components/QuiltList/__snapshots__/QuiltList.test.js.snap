// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`QuiltList renders properly 1`] = `
<div data-test-id="QUILT_LIST" class="wrapper">
  <p id="quiltListLabel" data-test-id="QUILT_LIST__TITLE" class="list-title">
    Select a quilt list:
  </p>
  <ul data-test-id="QUILT_LIST__LIST" role="listbox" tabindex="0" aria-labelledby="quiltListLabel" class="list">
    <quiltsizeoption-stub data="[object Object]" onclick="function onOptionClick() {
      return null;
    }" data-test-id="QUILT_LIST__OPTION"></quiltsizeoption-stub>
    <quiltsizeoption-stub data="[object Object]" isselected="true" onclick="function onOptionClick() {
      return null;
    }" data-test-id="QUILT_LIST__OPTION"></quiltsizeoption-stub>
    <quiltsizeoption-stub data="[object Object]" onclick="function onOptionClick() {
      return null;
    }" data-test-id="QUILT_LIST__OPTION"></quiltsizeoption-stub>
  </ul>
</div>
`;
